sensor/common/message/message.go

```diff
@@ -0,0 +1,49 @@
+package message
+
+import (
+	"context"
+
+	"github.com/stackrox/rox/generated/internalapi/central"
+)
+
+// ExpiringMessage is a wrapper on central.MsgFromSensor with the addition of a Context.
+// The context will be cancelled when the message is expired and should no longer be sent
+// to Central.
+type ExpiringMessage struct {
+	*central.MsgFromSensor
+	Context context.Context
+}
+
+// New creates an ExpiringMessage with msg and context.Background.
+func New(msg *central.MsgFromSensor) *ExpiringMessage {
+	return &ExpiringMessage{
+		MsgFromSensor: msg,
+		Context:       context.Background(),
+	}
+}
+
+// Wait is a helper function that will wait until the context on the message is Done.
+// This should be used when sending on blocking channels to abort the process the message
+// expired in the meantime. If the context is not set, this function blocks forever (i.e. wait on
+// context.Background() channel).
+func (m *ExpiringMessage) Wait() <-chan struct{} {
+	if m.Context == nil {
```

Comment:
[nit] I feel like this should never happen. The default value of `Context` is already `context.Background` and the only time when this is intended to be something else is when it's not `nil`. I think that maybe it would make sense to unexport the field `Context` from the `ExpiringMessage` to protect the code from this situation.

Reply:
> The default value of Context is already context.Background 

Do you mean the default value as in Go empty values? Because that's not the case `context.Context` is an interface, and it could be that someone created a `ExpiringMessage{}` somewhere by directly defining a struct and without assigning a context. This would result in a panic when `Wait()` is called. Playground example here: https://go.dev/play/p/KIY8WCltuL1

Reply:
I don't really like the setter method you proposed because it looks a lot like an option pattern function. Instead I've added a different constructor for these cases. Let me know what you think.

Reply:
> I've added a different constructor for these cases.

Makes sense. Thank you!